### PR TITLE
tlg1269.tlg002 initial conversion and markup

### LINKED ISSUE #2331: (urn:cts:greekLit:tlg1269.tlg002.1st1K-grc1.xml) new edition
Please see issue-2330 (https://github.com/OpenGreekAndLatin/First1KGreek/issues/2330) for zip file attachment

### PR DESCRIPTION
**Pull request by charhoughton**
fix #2331

### CODE REVIEW

**Comment by charhoughton** [comment]
#2331

**Comment by lcerrato** [comment]
@charhoughton 
The header credits for digital editors are order specific, so I see there were changes there.
I may not have time to look at this immediately but will as soon as possible. 
I'm curious why you removed the Latin tags in the last version?

**Comment by lcerrato** [comment]
@charhoughton 
Initial comments on the header:
I would not deviate from the TLG and spreadsheet title under most circumstances. I know this book differs a bit, but I don't think there is a reason to not use the spreadsheet title.
You added your own credits incorrectly. It needs to be name, org, role in that order.
The publication place, when modernized, is Stuttgart. Also, should read `<p>This pointer pattern extracts fragment.</p>` singular versus fragments.

I'm not sure of your structure here. For example, the header Vita et mores applies to fragments 463-481, so it should not be nested in fragment 463 only.

Unfortunately, I don't think that capturing the headers is going to work with this flat structure.

That opens up a whole bunch of other structure later on. Like, what happens at fragment 483?
https://archive.org/details/stoicorumveterum01arniuoft/page/108/mode/2up
That appears to be under two additional levels: A. Fragmenta logica et rhetorica and Utilitas Logicae.

So you would have 
Level 1 — A. Fragmenta logica et rhetorica
Level 2 — Utilitas Logicae
Level 3 — Fragment 483

which doesn't work because the start of the file has no identifiable Level 1, but it does have a Level 2:

Level 1 — ??
Level 2 — Vita et mores
Level 3 —  Fragment 463

**Comment by lcerrato** [comment]
@charhoughton 
I see some block quotes not captured. Be careful with things like this example the first fragment:

```
διὸ καὶ Τίμων περὶ αὐτοῦ φησιν οὕτως· (fr. 24 W.)
       <lb/>τίς δʼ οὗτος κτίλος ὣς ἐπιπωλεῖται στίχας ἀνδρῶν,
      <lb/>μωλυτὴς ἐπέων φίλος Ἄσσιος, ὅλμος ἄτολμος.<lb/>
καὶ σκωπτόμενος ὑπὸ τῶν συμμαθητῶν ἠνέσχετο καὶ ὄνος ἀκούων προσεδέχετο, 
```
This is a block quote not just  two lines indented.

```
διὸ καὶ Τίμων περὶ αὐτοῦ φησιν οὕτως· (fr. 24 W.)
        <quote rend="blockquote"><lb/>τίς δʼ οὗτος κτίλος ὣς ἐπιπωλεῖται στίχας ἀνδρῶν,
      <lb/>μωλυτὴς ἐπέων φίλος Ἄσσιος, ὅλμος ἄτολμος.</quote>
καὶ σκωπτόμενος ὑπὸ τῶν συμμαθητῶν ἠνέσχετο καὶ ὄνος ἀκούων προσεδέχετο, 
```

**Comment by lcerrato** [comment]
@charhoughton 
FYI I'm not big on the random indentation in the file. In general, we want something more readable. I think that Lace exports the files with this tidy format, but it's not the best for other work. I may reformat at some point here, but doing it now will create far too many differences.

I spotted a couple of random bad tag brackets:

(463) φασὶ δὲ καὶ Ἀντίγονον αὐτῷ τρισχιλq>ίας δοῦναι. (q> in the middle of the word)
(482) Diogenes Laërt. VII 41. ὁ δὲ Κλεάνθης ἓξ μέρη φησί type="subhead" rend="center">scil. (type="subhead" rend="center"> in the middle of the word)

Also, make sure to use em dashes instead of hyphens where appropriate. I

 see a few odd ones in Fragment 509. Note this is Latin, so we need to make sure that is captured. It also has two references.

Current version
`Philo de provid. II 48 (p. 79 Aucher). Age interim ponamus inter nos universum ingenitum ac sempiternum, iuxta illud quod suggerit sermo celeberrimorum philosophantium, sicut conscribunt Parmenides, Empedocles, Zeno, Cleanthes aliique divi homines ac velut verus quidam proprieque sacer coetus. Atqui ex ingenita materia quid miramur si pars aliqua generetur aut corrumpatur, partim per providentiam dei, partim ob rerum ordinem? Namque et ceterorum artificum unus quisque materiem generare haud solet, sed quam sibi sumit, eam figurat formatque et artis ordinem ei imponit. － － Quare iuxta huiusmodi hypothesin deus materiam primam non generavit sempiterne, sed materia ad usum sumpta, per eam caelum et terram atque species animalium plantarumque et omnia fecit － －. Quod nihilominus nullatenus impedit esse providentiam, etsi una cum materia mundus ingenitus supponatur. － 49. Quomodo? Quia non solum creare et edere materiam proprium est providentiae, verum etiam conservare moderarique, quod est factum.`

there are a couple of ways of handling this: tag the whole paragraph as Latin, which it is, or break it up: Note that the 49. here indicates a new reference.

`<p rend="indent"><bibl xml:lang="lat">Philo de provid. II 48 (p. 79 Aucher).</bibl> <foreign xml:lang="lat">Age interim ponamus inter nos universum ingenitum ac sempiternum, iuxta illud quod suggerit sermo celeberrimorum philosophantium, sicut conscribunt Parmenides, Empedocles, Zeno, Cleanthes aliique divi homines ac velut verus quidam proprieque sacer coetus. Atqui ex ingenita materia quid miramur si pars aliqua generetur aut corrumpatur, partim per providentiam dei, partim ob rerum ordinem? Namque et ceterorum artificum unus quisque materiem generare haud solet, sed quam sibi sumit, eam figurat formatque et artis ordinem ei imponit. — — Quare iuxta huiusmodi hypothesin deus materiam primam non generavit sempiterne, sed materia ad usum sumpta, per eam caelum et terram atque species animalium plantarumque et omnia fecit — —. Quod nihilominus nullatenus impedit esse providentiam, etsi una cum materia mundus ingenitus supponatur. — </foreign> <bibl>49.</bibl> <foreign xml:lang="lat">Quomodo? Quia non solum creare et edere materiam proprium est providentiae, verum etiam conservare moderarique, quod est factum.</foreign></p>`

or 

`<p rend="indent" xml:lang="lat"><bibl>Philo de provid. II 48 (p. 79 Aucher).</bibl> Age interim ponamus inter nos universum ingenitum ac sempiternum, iuxta illud quod suggerit sermo celeberrimorum philosophantium, sicut conscribunt Parmenides, Empedocles, Zeno, Cleanthes aliique divi homines ac velut verus quidam proprieque sacer coetus. Atqui ex ingenita materia quid miramur si pars aliqua generetur aut corrumpatur, partim per providentiam dei, partim ob rerum ordinem? Namque et ceterorum artificum unus quisque materiem generare haud solet, sed quam sibi sumit, eam figurat formatque et artis ordinem ei imponit. — — Quare iuxta huiusmodi hypothesin deus materiam primam non generavit sempiterne, sed materia ad usum sumpta, per eam caelum et terram atque species animalium plantarumque et omnia fecit — —. Quod nihilominus nullatenus impedit esse providentiam, etsi una cum materia mundus ingenitus supponatur. —  <bibl>49.</bibl> Quomodo? Quia non solum creare et edere materiam proprium est providentiae, verum etiam conservare moderarique, quod est factum.</p>`

In both cases the hyphens became em dashes.

File: data/tlg1269/tlg002/__cts__.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<ti:work xmlns:ti="http://chs.harvard.edu/xmlns/cts" groupUrn="urn:cts:greekLit:tlg1269" xml:lang="grc" urn="urn:cts:greekLit:tlg1269.tlg002">
+    <ti:title xml:lang="lat">Cleanthis Assii Fragmenta et Apophthegmata.</ti:title>
```
Comment (lcerrato) [inline data/tlg1269/tlg002/__cts__.xml:2]: I would stay with the spreadsheet on this. 
<title xml:lang="lat">Testimonia et fragmenta</title>

File: data/tlg1269/tlg002/__cts__.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    <ti:edition urn="urn:cts:greekLit:tlg1269.tlg002.1st1K-grc1" workUrn="urn:cts:greekLit:tlg1269.tlg002">
+        <ti:label xml:lang="lat">Cleanthis Assii Fragmenta et Apophthegmata.</ti:label>
```
Comment (lcerrato) [inline data/tlg1269/tlg002/__cts__.xml:4]: Same here. I don't think the book page is the way to go here. If in doubt, check with Alison.
<ti:label xml:lang="lat">Testimonia et fragmenta</ti:label>

No periods in either place.

File: data/tlg1269/tlg002/__cts__.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+        <ti:description xml:lang="mul">Cleanthes, Fragmenta and Apophthegmata, von Arnim, Teubner, 1903</ti:description>
```
Comment (lcerrato) [inline data/tlg1269/tlg002/__cts__.xml:5]: description should be 
**Common author name.** **Book title.** **Volume (if applicable).** **Editor Last, Editor First,** editor. **City: Publisher, Year.**

<ti:description xml:lang="mul">Cleanthes. Stoicorum veterum fragmenta, Vol. 1. von Arnim, Hans Friedrich August, editor. Stuttgart: Teubner, 1903. (1964 printing.)</ti:description>

**Review by lcerrato — CHANGES_REQUESTED** [review]

**Comment by charhoughton** [comment]
Hi @lcerrato, thanks for such in-depth comments! This file caused some issues for me, especially in the structure, so I wanted to have something completed and vaguely correct before adding them back in. For the first section in the text (without the header), would it be alright to label it as "0"? So it would look like:
 Level 1 — 0 (no header tag)
Level 2 — Vita et mores
Level 3 — Fragment 463

Secondly, would you be able to explain the difference between <bibl> and <foreign> when adding Latin? I have in the past used <foreign xml:lang="lat"> for everything, but noticed that was causing an Epidoc DTD failure in Travis recently. From what you showed me above, it looks like <foreign> gets used for instances in the text, but not when it's in references--is there anything else I should know about the usage of <foreign> or other applications when it isn't acceptable? 

Everything else seems very straightforward, and I will hopefully have this file in much better shape in the next week or so. Thanks again for everything!

**Comment by lcerrato** [comment]
@charhoughton 
To be clear, I would not add more structure. I would want there to be one level of reference, the fragments, as in TLG. Adding more structure serves no purpose here and makes for more complexity than is necessary.

Anything that is not obviously Greek has to be tagged as whatever language it is, otherwise it will be displayed as Greek and the parser will try to parse it as Greek.

We also want the references to be tagged. These are important to the text.

As I noted, you could make a case that some of these are English, but since most of the references include Latin abbreviations, I just use Latin.

Also, there is some obvious Latin here — full passages of it. 

I can take a look at the earlier pull request to see what might have broken. Whenever that happens you can just ask.

**Comment by lcerrato** [comment]
@charhoughton 
The previous commit broke because there were `<q>` tags nested in the `<foreign>` tags. You have to end the <foreign> tags and start the `<q xml:lang="lat">`.

As I mentioned, much of what you have as `<foreign>` is going to be `<bibl>`

`tlg1269.tlg002.1st1K-grc1.xml	 error: element "q" not allowed here [In (L261 C73); (L417 C96); (L419 C88); (L504 C119); (L504 C139); (L891 C81); (L967 C86); (L1050 C39); (L1066 C94); (L1066 C123); (L1071 C138); (L1071 C159); (L1200 C48)]`

There also appears to be run on text in 476 here:

Ad col. **XXVIIadnotat Buecheler:**

**Comment by charhoughton** [comment]
Okay, thanks for explaining. Basically, I should only use <foreign> if the Latin (or whatever language) isn't tagged as anything else, such as a quote or reference. And it sounds like with the flat fragment structure, I should just delete the heads because they don't accurately capture what they should be, yes?

**Comment by lcerrato** [comment]
(Sorry — just tag me so I'll see a notification. Otherwise I may miss any updates.)
@charhoughton 
Yes to the flat structure. You can delete the headers or comment them out. Preferably comment them out since they were already added.)

No, we need a language attribute anytime there is something other than Greek — it doesn't matter what that is. A quote or a reference still needs a language attribute. Remember, you want tags that impart meaning. So a reference in Latin, should be `<bibl xml:lang="lat">` or a Latin block quotation (--not something enclosed in quotation marks) should be `<quote rend="blockquote" xml:lang="lat">`. Latin enclosed in quotation marks can be `<q xml:lang="lat">`, etc.

If there is no other discernible meaning to the Latin — just running text — then you can just go to `<foreign xml:lang="lat">`

Please let me know if you have questions on particular passages as it's unlikely we'll spot any small issues in such a large file.

**Comment by charhoughton** [comment]
@lcerrato thanks so much. It also seems to me that we don't necessarily want to nest tags either; for example, it should be <foreign xml:lang="lat> random Latin text</foreign> <q xml:lang="lat">quote in Latin</q> <foreign xml:lang="lat"> more Latin... </foreign>rather than <foreign xml:lang="lat">random Latin text <q>quote in Latin</q>more Latin...</foreign>

**Comment by lcerrato** [comment]
@charhoughton 
Yes: I think the second example breaks in EpiDoc, so yes, there are some nesting rules. 
It's a pain because they can be valid XML and not pass the tests.

**Comment by charhoughton** [comment]
@lcerrato requested changes pushed through Travis; please let me know if they didn't get updated here or I still need to fix some things

**Comment by lcerrato** [comment]
@charhoughton 
This is a common problem:  a quotation which is indicated by formatting  is a `<quote>` versus indicating quotation marks `<q>`. If there are quotation marks being replaced, you need the latter. The former may or may not have quotation marks. If it does have quotation marks, you need both `<quote>` and `<q>`. Most commonly, these are mutually exclusive, and we look for quotations that are offset in the text, so `<quote rend="blockquote">`. We don't always read through for quotations in the Greek. (This can be done more easily in translation). 

(We also don't assign types to `<q>` tags, although that can be done, too.)

You have some `<q rend="blockquote">` — I am in the file now. Are all of these blockquotes that do not contain quotation marks? If so I will change them.

Also be careful — I see spaces missing in front of `<q>` tags. Think of it as the equivalent of a quotation mark so if there is a space in front of the quotation mark, you need a space in front of the tag, too.

**Comment by charhoughton** [comment]
@lcerrato fragment 472 has a blockquote with quotation marks; none of the others do. Thanks for clarifying, I'll be more careful about that in the future

**Comment by lcerrato** [comment]
@charhoughton 
No problem. Things are looking good!

I also see <lb/> at the end of the quotes/q — That's not needed. That's a carriage return, so you never need it at the close quote.

If you used <l></l> containers, you'd need to close that, of course.

**Comment by charhoughton** [comment]
@lcerrato can you elaborate on that last one a little? Something in my brain isn't quite clicking--I don't need "—" at the end of quotes?

**Comment by lcerrato** [comment]
@charhoughton 
Forgot to mark up the code! I edited the comment correctly this time. :)

**Comment by lcerrato** [comment]
@charhoughton The last steps are to update the pick list; archive the Trello card(s); and open a catalog pending issue for the new edition. https://github.com/PerseusDL/catalog_pending/issues/92
I've done all three!

All set!

**Comment by charhoughton** [comment]
@lcerrato thank you!!

**Pull request merged**